middlewares/authorizeUsersAndService.ts

```diff
@@ -0,0 +1,62 @@
+import jwt, { JwtPayload } from "jsonwebtoken";
+import { verifyCronJob, verifyDiscordBot } from "./authorizeBot";
+import { CustomRequest, CustomResponse } from "../types/global";
+import { NextFunction } from "express";
+import authenticate from "./authenticate";
+import authorizeRoles from "./authorizeRoles";
+const { Services } = require("../constants/bot");
+const ROLES = require("../constants/roles");
+const { INTERNAL_SERVER_ERROR_MESSAGE } = require("../constants/progresses");
+
+export const authorization = (allowedRoles: string[], allowedServices: string[]) => {
+  const rolesAreValid = allowedRoles.every((role) => Object.values(ROLES).includes(role));
+  if (!rolesAreValid) {
+    throw new Error("Invalid role");
+  }
+  const validServices = allowedServices.every((service) => Object.values(Services).includes(service));
```

Comment:
nit: `rolesAreValid` and `validServices` boolean variables name can be improved.

Reply:
Done